tdh/resource_user.go

```diff
@@ -194,23 +205,57 @@ func (r *userResource) Create(ctx context.Context, req resource.CreateRequest, r
 
 	// Generate API request body from plan
 	userRequest := customer_metadata.CreateUserRequest{
-		Usernames:    []string{plan.Email.ValueString()},
-		ServiceRoles: rolesReq,
+		Usernames: []string{plan.Email.ValueString()},
 	}
+	if r.client.Root.IsSre {
+		plan.Organizations.ElementsAs(ctx, &userRequest.Organizations, true)
+	}
+	if (r.client.Root.IsSre && !plan.Organizations.IsNull()) || !r.client.Root.IsSre {
+		userRequest.ServiceRoles = rolesReq
+		plan.PolicyIds.ElementsAs(ctx, &userRequest.PolicyIds, true)
+	}
+
 	plan.Tags.ElementsAs(ctx, &userRequest.Tags, true)
-	plan.PolicyIds.ElementsAs(ctx, &userRequest.PolicyIds, true)
-	if err := r.client.CustomerMetadata.CreateUser(&userRequest); err != nil {
-		resp.Diagnostics.AddError(
-			"Submitting request to create User",
-			"Could not create User, unexpected error: "+err.Error(),
-		)
-		return
+
+	if r.client.Root.IsSre {
+		if err := r.client.CustomerMetadata.CreateOrgOrSREUser(&userRequest); err != nil {
```

Comment:
handle it inside the client method, like you did for `GetUsers` since only reqUrl is different not request body, else it would be inconsistent